packages/sdk/src/types/runtime/assets/assetManager.ts

```diff
@@ -94,6 +97,71 @@ export class AssetManager {
         return createForwardPromise(asset, promise);
     }
 
+    public loadSoundAsset(groupName: string, uri: string, options: LoadSoundOptions): ForwardPromise<string> {
+
+        log.error('app', "STARTING LOADSOUNDASSET");
+        const soundAssetId = UUID();
+        this.enqueueLoadSoundPromise(
+            soundAssetId, {
+                resolve: () => { /* empty */ },
+                reject: () => { /* empty */ },
+            });
+        const promise = this.sendLoadAssetsPayload({

```

Comment:
>sendLoadAssetsPayload [](start = 29, length = 21)

This should utilize one of the existing asset messages, `load-assets` or `create-assets`. I don't think you need the `load-sound` message.

Possibly impacting: Looks like the `load-assets` codepath needs some adjustment to return a ForwardPromise<AssetGroup>.

Reply:
The trouble is that `load-assets` can return zero or more assets. We can't safely pre-generate IDs for the assets. That said, `createMaterial` and `createTexture` *do* return ForwardPromises, so adding a `createSound` would be consistent.

---
In reply to: [257468876](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/204#discussion_r257468876) [](ancestors = 257468876)

Reply:
The `create-asset` payload was designed for this use case.

---
In reply to: [258148811](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/204#discussion_r258148811) [](ancestors = 258148811,257468876)

Reply:
In fact, I think sound should *be* an asset unless there's compelling reason not to. There's a lot of code here that seems redundant with the existing asset pipeline (unload in particular).

---
In reply to: [258156474](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/204#discussion_r258156474) [](ancestors = 258156474,258148811,257468876)

Reply:
Yeah, a sound should be an asset. My main concern was it seems like unnecessary overhead per sound to have an assetgroup, but maybe that's not a big deal.